airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py

```diff
@@ -255,6 +258,8 @@ def custom_obj_api(self) -> CustomObjectsApi:
 
     def execute(self, context: Context):
         self.log.info("Creating sparkApplication.")
+        if self.override_existing:
+            self.on_kill()
```

Comment:
`spark_obj_spec` won't be available to delete the spark job?

Reply:
if you check the on_kill method, you find that it needs the launcher which is initiated just after this line; so your PR will not work.